See /kvmd/apps/otg/hid/keyboard.py for details
--- a/src/SingleReport/SingleAbsoluteMouse.cpp	2019-07-13 21:16:23.000000000 +0300
+++ b/src/SingleReport/SingleAbsoluteMouse.cpp	2019-10-05 00:58:34.592003332 +0300
@@ -43,7 +43,7 @@
     0x05, 0x01,                      /*     USAGE_PAGE (Generic Desktop) */
     0x09, 0x30,                      /*     USAGE (X) */
     0x09, 0x31,                      /*     USAGE (Y) */
-	0x16, 0x00, 0x80,				 /* 	Logical Minimum (-32768) */
+	0x16, 0x00, 0x00,				 /* 	Logical Minimum (0) */
 	0x26, 0xFF, 0x7F,				 /* 	Logical Maximum (32767) */
 	0x75, 0x10,						 /* 	Report Size (16), */
 	0x95, 0x02,						 /* 	Report Count (2), */
