https://github.com/arduino/Arduino/issues/6387
--- a/cores/arduino/USBCore.cpp	2019-09-20 15:48:38.000000000 +0300
+++ b/cores/arduino/USBCore.cpp	2020-11-11 19:56:49.233690476 +0300
@@ -375,8 +375,10 @@
 {
 	u8 i = setup.wIndex;
 
+#ifndef NO_SERIAL
 	if (CDC_ACM_INTERFACE == i)
 		return CDC_Setup(setup);
+#endif
 
 #ifdef PLUGGABLE_USB_ENABLED
 	return PluggableUSB().setup(setup);
@@ -466,7 +468,9 @@
 {
 	u8 interfaces = 0;
 
+#ifndef NO_SERIAL
 	CDC_GetInterface(&interfaces);
+#endif
 
 #ifdef PLUGGABLE_USB_ENABLED
 	PluggableUSB().getInterface(&interfaces);
